types/express-graphql/index.d.ts

```diff
@@ -80,32 +84,32 @@ declare namespace graphqlHTTP {
         /**
          * The parsed GraphQL document.
          */
-        document?: DocumentNode;
+        document: DocumentNode | null | undefined;
 
         /**
          * The variable values used at runtime.
          */
-        variables?: { [name: string]: any };
+        variables: { [name: string]: any } | null | undefined;
 
         /**
          * The (optional) operation name requested.
          */
-        operationName?: string;
+        operationName: string | null | undefined;
 
         /**
          * The result of executing the operation.
          */
-        result?: any;
+        result: any;
     }
 
     export interface GraphQLParams {
-        query?: string;
-        variables?: { [name: string]: any };
-        operationName?: string;
-        raw?: boolean;
+        query: string | null | undefined;
+        variables: { [name: string]: any } | null | undefined;
+        operationName: string | null | undefined;
+        raw: boolean | null | undefined;
     }
 
-    type Middleware = (request: Request, response: Response) => void;
+    type Middleware = (request: Request, response: Response) => Promise<undefined>;
```

Comment:
In https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L34 the RequestHandler is defined as
`(req: Request, res: Response, next: NextFunction): any;`
If this line has to be changed, I would vote for `any` as result type.

Reply:
thanks for review.
In [express-graphql](https://github.com/graphql/express-graphql/blob/v0.6.12/src/index.js#L123) the `Middleware` returns `Promise<void>`, compatible with `RequestHandler`'s return value.
Of course `any` is safe, but should we follow the definition of `express-graphql`?

Reply:
You're right. The definition should stick to the graphql-express rather than the general express types.